<!DOCTYPE html>
<html lang="zh-CN">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 6.3.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="Common Data File Formats in Bioinformatics 利用Sanger、illumina、第三代和第四代测序技术，我们测序得到的是带有质量值的碱基序列&#x3D;&#x3D;fastq&#x3D;&#x3D;格式，参考基因组是&#x3D;&#x3D;fasta&#x3D;&#x3D;格式。  用比对工具把fastq格式的序列回帖到对应的fasta格式的参考基因组序列">
<meta property="og:type" content="article">
<meta property="og:title" content="Common Data File Formats in Bioinformatics">
<meta property="og:url" content="http://example.com/2022/01/01/01Common%20Data%20File%20Formats%20in%20Bioinformatics/index.html">
<meta property="og:site_name" content="陈龙的小博客">
<meta property="og:description" content="Common Data File Formats in Bioinformatics 利用Sanger、illumina、第三代和第四代测序技术，我们测序得到的是带有质量值的碱基序列&#x3D;&#x3D;fastq&#x3D;&#x3D;格式，参考基因组是&#x3D;&#x3D;fasta&#x3D;&#x3D;格式。  用比对工具把fastq格式的序列回帖到对应的fasta格式的参考基因组序列">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://gitee.com/cling5899/Typora-img/raw/master/img/20220406150036.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914439.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914367.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914829.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914644.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914697.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914702.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914320.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131915893.png">
<meta property="article:published_time" content="2022-01-01T01:25:00.000Z">
<meta property="article:modified_time" content="2022-05-15T15:13:22.540Z">
<meta property="article:author" content="Long Chen">
<meta property="article:tag" content="Data File Formats">
<meta property="article:tag" content="生信数据格式">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://gitee.com/cling5899/Typora-img/raw/master/img/20220406150036.png">

<link rel="canonical" href="http://example.com/2022/01/01/01Common%20Data%20File%20Formats%20in%20Bioinformatics/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-CN'
  };
</script>

  <title>Common Data File Formats in Bioinformatics | 陈龙的小博客</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="切换导航栏">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">陈龙的小博客</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>首页</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>归档</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-CN">
    <link itemprop="mainEntityOfPage" href="http://example.com/2022/01/01/01Common%20Data%20File%20Formats%20in%20Bioinformatics/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Long Chen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="陈龙的小博客">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          Common Data File Formats in Bioinformatics
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">发表于</span>

              <time title="创建时间：2022-01-01 09:25:00" itemprop="dateCreated datePublished" datetime="2022-01-01T09:25:00+08:00">2022-01-01</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">更新于</span>
                <time title="修改时间：2022-05-15 23:13:22" itemprop="dateModified" datetime="2022-05-15T23:13:22+08:00">2022-05-15</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/%E8%AF%BE%E7%A8%8B%E7%AC%94%E8%AE%B0/" itemprop="url" rel="index"><span itemprop="name">课程笔记</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h1 id="Common-Data-File-Formats-in-Bioinformatics"><a href="#Common-Data-File-Formats-in-Bioinformatics" class="headerlink" title="Common Data File Formats in Bioinformatics"></a>Common Data File Formats in Bioinformatics</h1><ul>
<li><p>利用<strong>Sanger、illumina、第三代和第四代测序技术</strong>，我们测序得到的是带有质量值的碱基序列&#x3D;&#x3D;fastq&#x3D;&#x3D;格式，参考基因组是&#x3D;&#x3D;fasta&#x3D;&#x3D;格式。</p>
</li>
<li><p>用比对工具把fastq格式的序列回帖到对应的fasta格式的参考基因组序列，就可以产⽣&#x3D;&#x3D;sam&#x3D;&#x3D;格式的⽐对⽂件。</p>
</li>
<li><p>把sam格式的⽂本⽂件压缩成&#x3D;&#x3D;⼆进制bam&#x3D;&#x3D;⽂件可以节省空间。</p>
</li>
<li><p>如果是记录某些位点或者区域碱基的变化，就是&#x3D;&#x3D;VCF&#x3D;&#x3D;⽂件格式。</p>
</li>
<li><p>如果对参考基因组上面的各个区段标记它们的性质，比如哪些区域是外显子，内含子， UTR等等，这就是&#x3D;&#x3D;gtf&#x2F;gff&#x3D;&#x3D;格式。</p>
</li>
<li><p>如果只是为了单纯描述某个基因组区域，就是&#x3D;&#x3D;bed&#x3D;&#x3D;格式⽂件，记录染色体号以及起始终止坐标，正负链即可。</p>
</li>
</ul>
<h2 id="fastq文件"><a href="#fastq文件" class="headerlink" title="fastq文件"></a>fastq文件</h2><h3 id="概念"><a href="#概念" class="headerlink" title="概念"></a>概念</h3><blockquote>
<p>fastq&#x3D;fasta+quality(Fasta格式加上质量数据)</p>
</blockquote>
<p><a target="_blank" rel="noopener" href="https://support.illumina.com/bulletins/2016/04/chinese-fastq-files-explained.html">https://support.illumina.com/bulletins/2016/04/chinese-fastq-files-explained.html</a></p>
<p><a target="_blank" rel="noopener" href="https://help.basespace.illumina.com/articles/descriptive/fastq-files/">https://help.basespace.illumina.com/articles/descriptive/fastq-files/</a></p>
<blockquote>
<p>Illumina测序技术使用 <a target="_blank" rel="noopener" href="https://v.youku.com/v_show/id_XMzMxMTk0NzQ5Ng==.html?spm=a2h3j.8428770.3416059.1">簇生成和边合成边测序（SBS）化学技术</a>对流动槽（flow cell）上数百万或数十亿簇（cluster）进行测序，具体簇的数目取决于测序平台。 在边合成边测序化学过程中，仪器上的实时分析（RTA）软件对每个簇的每个循环进行碱基检出和存储。 RTA以单个读取碱基（base call，或称BCL）文件的形式存储碱基检出数据。 测序完成后，必须将BCL文件中的测定的碱基转换为序列数据。 此过程称为BCL到FASTQ的转换。</p>
</blockquote>
<p>FASTQ是基于文本的，保存生物序列（通常是核酸序列）和其测序质量信息的<strong>标准</strong>格式。其序列以及质量信息都是使用一个ASCII字符标示，最初由Sanger开发，目的是<strong>将FASTA序列与质量数据放到一起</strong>，目前已经成为高通量测序结果的事实标准。</p>
<p>对于单端测序的运行，将为每个流动槽上每条通道的每个样品创建一个Read 1（R1）FASTQ文件。 对于双端测序的运行，将为每个流动槽上每条通道的每个样品各创建一个R1和一个Read 2（R2）FASTQ文件。 FASTQ文件是使用扩展名*.fastq.gz压缩和创建的。</p>
<h3 id="格式"><a href="#格式" class="headerlink" title="格式"></a>格式</h3><p><img src="https://gitee.com/cling5899/Typora-img/raw/master/img/20220406150036.png" alt="img"></p>
<p><img src="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914439.png" alt="preload"></p>
<ol>
<li><p>第一行以@开头，序列标识符，其中包含有关测序运行和簇的信息。 该行的具体内容会因使用的BCL到FASTQ转换软件而不同。（与FASTA格式的描述行类似）</p>
</li>
<li><p>序列（碱基信号； A，C，T，G和N）。</p>
</li>
<li><p>分隔符，只是一个加号（+），之后可以再次加上序列的标识及描述信息（可选）。</p>
</li>
<li><p>读取碱基的质量值。，与第二行的序列相对应，长度必须与第二行相同。这些是Phred +33编码的，使用ASCII字符表示数字质量值。</p>
</li>
</ol>
<p>其中!为最低质量、~则为最高质量。以下字符从左到右代表从低到高的质量得分的：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">!&quot;#$%&amp;&#x27;()*+,-./0123456789:;&lt;=&gt;?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz&#123;|&#125;~</span><br></pre></td></tr></table></figure>

<p><img src="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914367.png" alt="image-20220406150612091"></p>
<p>质量值的计算公式如下：$Q&#x3D;-10 \times log_{10}P$</p>
<p>这个公式中Q表示的是潜在错误的概率，Q值越大，表明测错的可能性越小。下面是Q值对应的错误率表：</p>
<table>
<thead>
<tr>
<th>Phred Quality Score</th>
<th>Probability of incorrect base call</th>
<th>Base call accuracy</th>
</tr>
</thead>
<tbody><tr>
<td>10</td>
<td>1 in 10</td>
<td>90%</td>
</tr>
<tr>
<td>20</td>
<td>1 in 100</td>
<td>99%</td>
</tr>
<tr>
<td>30</td>
<td>1 in 1000</td>
<td>99.9%</td>
</tr>
<tr>
<td>40</td>
<td>1 in 10000</td>
<td>99.99%</td>
</tr>
<tr>
<td>50</td>
<td>1 in 100000</td>
<td>99.999%</td>
</tr>
<tr>
<td>60</td>
<td>1 in 1000000</td>
<td>99.9999%</td>
</tr>
</tbody></table>
<p>举个犊子，假设&#x3D;&#x3D;质量信息是5&#x3D;&#x3D;，这个5看成字符，不要看成数字，那么他对应的十进制数就是53，如果使用的是Sanger测序，那么质量评分采用的就是Phred+33的形式，那么Q &#x3D; 53-33&#x3D;20&#x3D;-10lg(errorP)，所以errorP &#x3D; 0.01。也就计算出错误率啦，就便于我们进行质控。每一个碱基都有一个质量评分，所以第2行和第4行的位数是相同的。&#x3D;&#x3D;Q值&#x3D;20&gt;&gt;errorP&#x3D;0.01&gt;&gt;碱基准确度就是99%。&#x3D;&#x3D;</p>
<p><img src="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914829.png" alt="image-20220406153327147"></p>
<p><img src="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914644.png" alt="preload"></p>
<blockquote>
<p>质量评分指的是一个碱基的错误概率的对数值。其最初在Phred拼接软件中定义与使用，对于每个碱基的质量编码标示，不同的软件采用不同的方案，目前有5种方案：</p>
</blockquote>
<ul>
<li>Sanger，Phred quality score，值的范围从0到92，对应的ASCII码从33到126，但是对于测序数据（raw read data）质量得分通常小于60，序列拼接或者mapping可能用到更大的分数。</li>
<li>Solexa&#x2F;Illumina 1.0, Solexa&#x2F;Illumina quality score，值的范围从-5到63，对应的ASCII码从59到126，对于测序数据，得分一般在-5到40之间；</li>
<li>Illumina 1.3+，Phred quality score，值的范围从0到62对应的ASCII码从64到126，低于测序数据，得分在0到40之间；</li>
<li>Illumina 1.5+，Phred quality score，但是0到2作为另外的标示；</li>
<li>Illumina 1.8+ ，Phred quality score。</li>
</ul>
<h2 id="fasta文件"><a href="#fasta文件" class="headerlink" title="fasta文件"></a>fasta文件</h2><h3 id="概念-1"><a href="#概念-1" class="headerlink" title="概念"></a>概念</h3><p>FASTA格式是一种用于表示&#x3D;&#x3D;核苷酸序列&#x3D;&#x3D;或&#x3D;&#x3D;多肽序列&#x3D;&#x3D;的文本格式。其中碱基对或氨基酸用&#x3D;&#x3D;单个字母&#x3D;&#x3D;来表示，且允许在序列前添加序列名及注释。该格式已成为生物信息学领域的一项标准。</p>
<p>FASTA简明的格式降低了序列操纵和分析的难度，令序列可被文本处理工具和诸如<a target="_blank" rel="noopener" href="https://zh.wikipedia.org/wiki/Python">Python</a>、<a target="_blank" rel="noopener" href="https://zh.wikipedia.org/wiki/Ruby">Ruby</a>和<a target="_blank" rel="noopener" href="https://zh.wikipedia.org/wiki/Perl">Perl</a>等<a target="_blank" rel="noopener" href="https://zh.wikipedia.org/wiki/%E8%84%9A%E6%9C%AC%E8%AF%AD%E8%A8%80">脚本语言</a>处理。</p>
<h3 id="格式-1"><a href="#格式-1" class="headerlink" title="格式"></a>格式</h3><p>FASTA格式中的一条完整序列，包含开头的单行描述行和多行序列数据。描述行行首前置半角大于号（“&gt;”）以和数据行区分。“&gt;”后紧接的内容为该序列的标识符（&#x3D;&#x3D;唯一的&#x3D;&#x3D;），该行剩余部分则为序列的描述（标识符与描述均非必须）。“&gt;”和标识符之间不应有空格，且建议将<strong>单行内容限制在80字符以内</strong>。序列的结束以下一条序列的“&gt;”出现为标识。如下为FASTA评论区格式一条序列的示例：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">&gt;gi|31563518|ref|NP_852610.1 关，瓦斯4 microtubule-associated proteins 1A/1B light chain 3A isoform b [Homo sapiens]</span><br><span class="line">MKMRFFSSPCGKAAVDPADRCKEVQQIRDQHPSKIPVIIERYKGEKQLPVLDKTKFLVPDHVNMSELVKI</span><br><span class="line">IRRRLQLNPTQAFFLLVNQHSMVSVSTPIADIYEQEKDEDGFLYMVYASQETFGFIRENE</span><br></pre></td></tr></table></figure>

<p><img src="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914697.png" alt="img"></p>
<p><strong>核苷酸序列：</strong></p>
<p><img src="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914702.png" alt="preload"></p>
<p><strong>氨基酸序列：</strong></p>
<p><img src="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131914320.png" alt="img"></p>
<h2 id="SAM文件-x2F-BAM文件"><a href="#SAM文件-x2F-BAM文件" class="headerlink" title="SAM文件&#x2F;BAM文件"></a>SAM文件&#x2F;BAM文件</h2><h3 id="概念-2"><a href="#概念-2" class="headerlink" title="概念"></a>概念</h3><p><a target="_blank" rel="noopener" href="https://www.samformat.info/sam-format-flag">官网介绍</a></p>
<p><a target="_blank" rel="noopener" href="https://www.cnblogs.com/dataanaly/p/13177364.html">SAM详细介绍</a></p>
<p><a target="_blank" rel="noopener" href="https://blog.csdn.net/genome_denovo/article/details/78712972">CSDN上的SAM介绍</a></p>
<p>当我们测序得到的fastq数据map到基因组（比如用bowtie2来mapping）之后，会得到一个以sam或bam为扩展名的文件。这里，SAM的全称是<strong>sequence alignment&#x2F;map format</strong>。而BAM就是SAM的二进制文件，也就是压缩格式的sam文件。</p>
<p>SAM格式文件包括<strong>头部注释部分</strong>和<strong>比对结果部分</strong>，头部分为’’可选部分’’。头部分位于比对部分之前，以“@”开头。比对部分有11列是固定的，其他多列可选。</p>
<p>头部信息介绍</p>
<p>每个标题行以字符”@”开头，后面是两个字母的记录类型代码。在标题中，每一行都是由制表符分隔的。除了@CO行，每个数据字段都遵循格式”TAG:VALUE”，其中TAG是两个字母组成的字符串，定义了内容和值的格式。每个标题行应该匹配:&#x2F;^ @(HD|SQ|RG|PG)(t[A-Z&#x2F;a-z][A-Z&#x2F;a-z&#x2F;0-9]:[-~]+)+$ &#x2F;或&#x2F;^@CO\t.*&#x2F;。小写字母是保留给用户自定义内容使用的，不会在任何版本定义中出现。以下给出了定义的记录类型和标记。<strong>当以下TAG标签出现”*“时，表示该TAG是必须的，如@SQ必须有SN和LN字段</strong>。</p>
<p>比对信息介绍<br>比对信息部分(alignment section)，每一行表示一个片段（segment）的比对信息，包括11个必须的字段（mandatory fields）和一个可选的字段，字段之间用tab分割。必须的字段有11个，顺序固定，不可用时，根据字段定义，可以为’0’或者’*’，这是11个字段包括：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line">1. QNAME 比对片段的（template）的编号；read name，read的名字通常包括测序平台等信息</span><br><span class="line"># eg.E00528:375:HN2JTCCXY:8:1123:5457:24479</span><br><span class="line">2. FLAG  位标识，template mapping情况的数字表示，每一个数字代表一种比对情况，这里的值是符合情况的数字相加总和</span><br><span class="line"># eg.16</span><br><span class="line">3. RNAME 参考序列的编号，如果头部注释中对SQ-SN进行了定义，这里必须和其保持一致，另外对于没有mapping上的序列，这里是&#x27;*&#x27;；</span><br><span class="line"># eg.chr1</span><br><span class="line">4. POS  比对上的位置，注意是从1开始计数，没有比对上，此处为0;</span><br><span class="line"># eg.36576599</span><br><span class="line">5. MAPQ mapping的质量，比对的质量分数，越高说明该read比对到参考基因组上的位置越唯一;</span><br><span class="line"># eg.42</span><br><span class="line">6. CIGAR 简要比对信息表达式（Compact Idiosyncratic Gapped Alignment Report），其以参考序列为基础，使用数字加字母表示比对结果，match/mismatch、insertion、deletion 对应字母 M、I、D。比如3S6M1P1I4M，前三个碱基被剪切去除了，然后6个比对上了，然后打开了一个缺口，有一个碱基插入，最后是4个比对上了，是按照顺序的；</span><br><span class="line"># eg.36M 表示36个碱基在比对时完全匹配</span><br><span class="line"></span><br><span class="line">###注：第七列到第九列是mate的信息，若是单末端测序这几列均无意义###</span><br><span class="line"></span><br><span class="line">7. RNEXT 配对片段（即mate）比对上的参考序列的编号，没有另外的片段，这里是&#x27;*&#x27;，同一个片段，用&#x27;=&#x27;；</span><br><span class="line"># eg.*</span><br><span class="line">8. PNEXT 配对片段（即mate）比对到参考序列上的第一个碱基位置，若无mate,则为0；</span><br><span class="line"># eg.0</span><br><span class="line">9. TLEN  Template（文库插入序列）的长度，最左边的为正，最右边的为负，中间的不用定义正负，不分区段（single-segment)的比对上，或者不可用时，此处为0(ISIZE，Inferred fragment size.详见Illumina中paired end sequencing 和 mate pair sequencing，是负数，推测应该是两条read之间的间隔(待查证)，若无mate则为0);</span><br><span class="line"># eg.0</span><br><span class="line">10. SEQ  序列片段的序列信息，如果不存储此类信息，此处为&#x27;*&#x27;，注意CIGAR中M/I/S/=/X对应数字的和要等于序列长度；</span><br><span class="line"># eg.CGTTTCTGTGGGTGATGGGCCTGAGGGGCGTTCTCN</span><br><span class="line">11. QUAL 序列的质量信息, read质量的ASCII编码。格式同FASTQ一样。</span><br><span class="line"># eg.F-?DD&lt;,EEFFEEC?F5FA7FAF8FAF;F@E:DF&gt;FEEAFFFFFCFFC4EFDFDDFEFE=EFFF</span><br><span class="line">12.第十二列及之后：Optional fields，以tab建分割。</span><br><span class="line"># eg.AS:i:-1 XN:i:0 XM:i:1 XO:i:0 XG:i:0 NM:i:1 MD:Z:35T0 YT:Z:UU</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">\1. **QNAME**：reads名称。</span><br><span class="line">\2. **FLAG**：reads比对情况。不同的情况对应不同的值，这里的数字是所有情况的和。</span><br><span class="line">\3. **RNAME**：比对至参考序列的名称。</span><br><span class="line">\4. **POS**：比对到的位置。</span><br><span class="line">\5. **MAPQ**：比对质量。</span><br><span class="line">\6. **CIGAR**：比对情况信息。</span><br><span class="line">\7. **RNEXT**：与之配对的另一条reads所在的参考序列名称。&quot;=&quot;表示位于同一个参考序列上，&quot;*&quot;表示没有另一条reads。</span><br><span class="line">\8. **PNEXT**：与之配对的另一条reads所在的位置。</span><br><span class="line">\9. **TLEN**：插入片段长度。</span><br><span class="line">\10. **SEQ**：reads序列。</span><br><span class="line">\11. **QUAL**：reads序列质量</span><br><span class="line">NH:i:n 表示reads比对到参考序列位置的个数。</span><br><span class="line">AS:i:n 表示比对得分。</span><br></pre></td></tr></table></figure>

<h3 id="格式-2"><a href="#格式-2" class="headerlink" title="格式"></a>格式</h3><p>例</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">@HD VN:1.4	SO:coordinate</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">@HD     VN:1.5  SO:unsorted     GO:query  ##输出文件第一行</span><br><span class="line">@SQ     SN:k127_19953   LN:526</span><br><span class="line">@SQ     SN:k127_3992    LN:562     ###@SQ 参考序列字典，@SQ行的顺序定义了比对文件排序顺序。</span><br><span class="line">@SQ     SN:k127_15965   LN:635     ###SN* 参考序列名字(染色体)。用于RNAME和PNEXT比对的记录</span><br><span class="line">@SQ     SN:k127_32914   LN:521     ###LN* 参考序列长度，范围:[$1,2^&#123;31&#125;-1$]</span><br><span class="line">@SQ     SN:k127_35905   LN:622     ###</span><br><span class="line">@SQ     SN:k127_7984    LN:664     ###</span><br><span class="line">@SQ     SN:k127_17959   LN:506     ###</span><br><span class="line">@SQ     SN:k127_20950   LN:547     ###</span><br><span class="line">@SQ     SN:k127_11976   LN:697     ###</span><br><span class="line">@SQ     SN:k127_26932   LN:770     ###</span><br><span class="line">@SQ     SN:k127_2995    LN:515     ###</span><br><span class="line">@SQ     SN:k127_23941   LN:826     ###</span><br><span class="line">@SQ     SN:k127_41887   LN:788     ###</span><br><span class="line">@SQ     SN:k127_14968   LN:970     ###</span><br><span class="line">@SQ     SN:k127_8983    LN:511     ###</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"># 例如</span><br><span class="line">@SQ	SN:chr1	LN:249250621	M5:1b22b98cdeb4a9304cb5d48026a85128	UR:file:/opt/biosoft/ref/sv/hg19_ref_genome.primary-assembly.fa</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">E00514:173:H3C3JCCXY:4:1124:12398:67234    337 Chr00   32904   0   150M    Chr09   33498107    0   TCAATTTCACTTGAAGCTTACTTGTAGTTTCAGGCTTGGTCAAGCGCGATACAAACCATGTAGTAGGAGTCCTCCAAGTCGCCAAGCTAGGGGATCTGCTGAAAGAGGTGACAGACAAGGTAAGCAATCAGAGCTCTAAGCAATCAGTCC  iieiiiii`eiiiiiiiiiiiiiiieiiiiiiiieiiiiiiiiiiiiiiiiiiiiieiiiiiiiiiiiiieiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiieiiiiiiiii`iiieeieieeieee``  AS:i:-6 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:136C13 YT:Z:UU NH:i:8  CC:Z:Chr10  CP:i:18604313   HI:i:0  RG:Z:J36CK1</span><br><span class="line">E00514:173:H3C3JCCXY:4:1124:12398:67234    369 Chr00   32904   0   150M    Chr16   2469225 0   TCAATTTCACTTGAAGCTTACTTGTAGTTTCAGGCTTGGTCAAGCGCGATACAAACCATGTAGTAGGAGTCCTCCAAGTCGCCAAGCTAGGGGATCTGCTGAAAGAGGTGACAGACAAGGTAAGCAATCAGAGCTCTAAGCAATCAGTCC  iieiiiii`eiiiiiiiiiiiiiiieiiiiiiiieiiiiiiiiiiiiiiiiiiiiieiiiiiiiiiiiiieiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiieiiiiiiiii`iiieeieieeieee``  AS:i:-6 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:136C13 YT:Z:UU NH:i:8  CC:Z:Chr10  CP:i:18604313   HI:i:2  RG:Z:J36CK1</span><br><span class="line">E00514:173:H3C3JCCXY:4:1124:12398:67234    369 Chr00   32904   0   150M    Chr16   29515410    0   TCAATTTCACTTGAAGCTTACTTGTAGTTTCAGGCTTGGTCAAGCGCGATACAAACCATGTAGTAGGAGTCCTCCAAGTCGCCAAGCTAGGGGATCTGCTGAAAGAGGTGACAGACAAGGTAAGCAATCAGAGCTCTAAGCAATCAGTCC  iieiiiii`eiiiiiiiiiiiiiiieiiiiiiiieiiiiiiiiiiiiiiiiiiiiieiiiiiiiiiiiiieiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiieiiiiiiiii`iiieeieieeieee``  AS:i:-6 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:136C13 YT:Z:UU NH:i:8  CC:Z:Chr10  CP:i:18604313   HI:i:4  RG:Z:J36CK1</span><br></pre></td></tr></table></figure>

<h2 id="GTF和GFF文件"><a href="#GTF和GFF文件" class="headerlink" title="GTF和GFF文件"></a>GTF和GFF文件</h2><h3 id="概念-3"><a href="#概念-3" class="headerlink" title="概念"></a>概念</h3><p>GTF全称是Gene transfer format，用于储存基因结构信息，总共有9列。关于gtf文件，我在文章：<strong>生信中各种ID转换</strong>中就已经有所应用。我之前在<strong>TCGA</strong><a target="_blank" rel="noopener" href="https://cloud.tencent.com/solution/database?from=10680"><strong>数据库</strong></a><strong>差异分析</strong>的文章中，也是通过gtf文件进行ID转换的。</p>
<h3 id="格式-3"><a href="#格式-3" class="headerlink" title="格式"></a>格式</h3><p><img src="https://raw.githubusercontent.com/Cling5899/Personal_Typora_img/master/img/202205131915893.png" alt="preload"></p>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/Data-File-Formats/" rel="tag"># Data File Formats</a>
              <a href="/tags/%E7%94%9F%E4%BF%A1%E6%95%B0%E6%8D%AE%E6%A0%BC%E5%BC%8F/" rel="tag"># 生信数据格式</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item"></div>
      <div class="post-nav-item">
    <a href="/2022/02/01/Biostatistics-chapter1-5/" rel="next" title="Biostatistics课程学习笔记Chapter1-5">
      Biostatistics课程学习笔记Chapter1-5 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          文章目录
        </li>
        <li class="sidebar-nav-overview">
          站点概览
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#Common-Data-File-Formats-in-Bioinformatics"><span class="nav-number">1.</span> <span class="nav-text">Common Data File Formats in Bioinformatics</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#fastq%E6%96%87%E4%BB%B6"><span class="nav-number">1.1.</span> <span class="nav-text">fastq文件</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%A6%82%E5%BF%B5"><span class="nav-number">1.1.1.</span> <span class="nav-text">概念</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%A0%BC%E5%BC%8F"><span class="nav-number">1.1.2.</span> <span class="nav-text">格式</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#fasta%E6%96%87%E4%BB%B6"><span class="nav-number">1.2.</span> <span class="nav-text">fasta文件</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%A6%82%E5%BF%B5-1"><span class="nav-number">1.2.1.</span> <span class="nav-text">概念</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%A0%BC%E5%BC%8F-1"><span class="nav-number">1.2.2.</span> <span class="nav-text">格式</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#SAM%E6%96%87%E4%BB%B6-x2F-BAM%E6%96%87%E4%BB%B6"><span class="nav-number">1.3.</span> <span class="nav-text">SAM文件&#x2F;BAM文件</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%A6%82%E5%BF%B5-2"><span class="nav-number">1.3.1.</span> <span class="nav-text">概念</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%A0%BC%E5%BC%8F-2"><span class="nav-number">1.3.2.</span> <span class="nav-text">格式</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#GTF%E5%92%8CGFF%E6%96%87%E4%BB%B6"><span class="nav-number">1.4.</span> <span class="nav-text">GTF和GFF文件</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%A6%82%E5%BF%B5-3"><span class="nav-number">1.4.1.</span> <span class="nav-text">概念</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%A0%BC%E5%BC%8F-3"><span class="nav-number">1.4.2.</span> <span class="nav-text">格式</span></a></li></ol></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Long Chen</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">20</span>
          <span class="site-state-item-name">日志</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
        <span class="site-state-item-count">9</span>
        <span class="site-state-item-name">分类</span>
      </div>
      <div class="site-state-item site-state-tags">
        <span class="site-state-item-count">55</span>
        <span class="site-state-item-name">标签</span>
      </div>
  </nav>
</div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2023</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Long Chen</span>
</div>
  <div class="powered-by">由 <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a> 强力驱动
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
